Autopilot/Inc/SensorFusion.hpp

```diff
@@ -0,0 +1,62 @@
+/**
+ * Sensor Fusion - Converts raw sensor data into human readable formatted structs
+ * Author: Lucy Gong, Dhruv Rawat, Anthony Bertnyk
+ */
+
+#include "AttitudeDatatypes.hpp"
+#include "fetchSensorMeasurementsMode.hpp"
+
+#ifndef SENSORFUSION_HPP
+#define SENSORFUSION_HPP
+
+// -1 = FAILED
+// 0 = SUCCESS
+// 1 = Old Data
+struct SFError_t{
+    int errorCode;
+};
+
+struct SFOutput_t {
+    float IMUroll, IMUpitch, IMUyaw; //rad
+    float IMUrollrate, IMUpitchrate, IMUyawrate; //rad/s
+    float Airspeed; //m/s
+    float altitude; //m
+    float altSpeed; //m/s
+    long double latitude; //Decimal degrees
+    float latSpeed; //m/s
+    long double longitude; //Decimal degrees
+    float longSpeed; //m/s
+};
+
+/**
+ * Initiate sensor fusion.
+ */ 
+SFError_t SF_Init(void);
+
+/**
+ * Get fused sensor data.
+ * @param[out]    output  Reference to an output struct for fused data.
+ */ 
+SFError_t SF_GetResult(SFOutput_t *output);
```

Comment:
Something to consider: 

- It may be better to split SF_execute() functions such that each one executes one specific algorithm. That way Pathmanager doesn't run the Madgwick algorithm and all.

Reply:
The Kalman filter can actually make use of output from the Madgwick algorithm so I was thinking of combining them to run one after the other. But since Madgwick can run on its own it might be worth splitting them anyways.

Reply:
Should weigh in here. There's a problem with this design because path manager and attitude manager are both calling execute at different rates. So you end up in this case where execute would no longer get called at regular intervals.

I didn't anticipate that the measurements would be so coupled though, so it's a pain to share data between path and attitude.

Let's talk about this a bit, cuz it's an important consideration, but one thing that can be done is sensor fusion can live in its own thread, and just compute things at whatever constant interval you like (probs as fast as the fastest sensor). And path manager/attitude manager, rather than calling a function that actually computes measurements, can call functions that retrieve the latest measurements.